qiskit/transpiler/passes/layout/disjoint_utils.py

```diff
@@ -127,6 +127,21 @@ def combine_barriers(dag: DAGCircuit, retain_uuid: bool = True):
                 node.op.label = None
 
 
+def check_layout_isolated_to_component(dag: DAGCircuit, coupling_map: CouplingMap) -> bool:
+    """Check that the layout of the dag does not require connectivity across connected components
+    in the CouplingMap"""
+    qubit_indices = {bit: index for index, bit in enumerate(dag.qubits)}
+    component_sets = [set(x.graph.nodes()) for x in coupling_map.connected_components()]
+    for inst in dag.two_qubit_ops():
+        component_index = None
+        for i in range(len(component_sets)):
```

Comment:
Is there a reason not to use `enumerate` here, since we want to use both the index and the value in this loop?

Reply:
Oh good catch, yeah I should just use enumerate here instead of `range()`

Reply:
Updated in: https://github.com/Qiskit/qiskit-terra/pull/9840/commits/905bcaf145178931b4059874d8b36194e86a0c86